imap/util.c

```diff
@@ -784,20 +784,16 @@ void imap_error(const char *where, const char *msg)
  * to "/".  IMAP servers are not required to do this.
  * Moreover, IMAP servers may dislike the path ending with the delimiter.
  */
-char *imap_fix_path(struct ImapAccountData *adata, const char *mailbox, char *path, size_t plen)
+char *imap_fix_path(char delim, const char *mailbox, char *path, size_t plen)
 {
   int i = 0;
-  char delim = '\0';
-
-  if (adata)
-    delim = adata->delim;
 
   while (mailbox && *mailbox && i < plen - 1)
   {
     if ((ImapDelimChars && strchr(ImapDelimChars, *mailbox)) || (delim && *mailbox == delim))
     {
       /* use connection delimiter if known. Otherwise use user delimiter */
-      if (!adata)
+      if (!delim)
```

Comment:
Is this logically equivalent? My (limited) understanding of the `\0` character comparisons leans me towards no. Does that work or is a comparison to `\0` needed?

I apologise for the newb question.

Reply:
It works because the char '\0' is 0 as integer. But I will make it explicit.